From ae523a2d8262aee255702f3046fe59529a89f2ef Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Tue, 17 Feb 2026 14:44:55 +0000
Subject: [PATCH 2/9] Disable secure app spawning by default

Exec-based app spawning breaks root access by re-exec'ing app processes
into a fresh app_process binary, which loses the modified Zygote
environment that root solutions depend on.

Change the default for persist.security.exec_spawn from true to false.
Users can still enable it manually in Settings > Security > Exploit
Protection if they don't need root.

Change-Id: I0000000000000000000000000000000000000014
---
 core/java/android/ext/settings/ExtSettings.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/android/ext/settings/ExtSettings.java b/core/java/android/ext/settings/ExtSettings.java
index 821f3d238..cb58a97f9 100644
--- a/core/java/android/ext/settings/ExtSettings.java
+++ b/core/java/android/ext/settings/ExtSettings.java
@@ -22,7 +22,7 @@ import java.util.function.ToIntFunction;
 public class ExtSettings {

     public static final BoolSysProperty EXEC_SPAWNING = new BoolSysProperty(
-            "persist.security.exec_spawn", true);
+            "persist.security.exec_spawn", false);

     public static final BoolSetting ALLOW_KEYGUARD_CAMERA = new BoolSetting(
             Setting.Scope.SYSTEM_PROPERTY, "persist.keyguard.camera", true);
-- 
2.53.0

